source/Octopus.Cli/Commands/Deployment/DeploymentCommandBase.cs

```diff
@@ -81,11 +82,68 @@ protected override async Task ValidateParameters()
             if (string.IsNullOrWhiteSpace(ProjectName)) throw new CommandException("Please specify a project name using the parameter: --project=XYZ");
             if (IsTenantedDeployment && DeployToEnvironmentNames.Count > 1) throw new CommandException("Please specify only one environment at a time when deploying to tenants.");
             if (Tenants.Contains("*") && (Tenants.Count > 1 || TenantTags.Count > 0)) throw new CommandException("When deploying to all tenants using --tenant=* wildcard no other tenant filters can be provided");
-
             if (IsTenantedDeployment && !await Repository.SupportsTenants().ConfigureAwait(false))
                 throw new CommandException("Your Octopus Server does not support tenants, which was introduced in Octopus 3.4. Please upgrade your Octopus Server, enable the multi-tenancy feature or remove the --tenant and --tenanttag arguments.");
+            /*
+             * A create release operation can also optionally deploy the release, however any invalid options that
+             * are specific only to the deployment will fail after the release has been created. This can leave
+             * a deployment in a half finished state, so this validation ensures that the input relating to the
+             * deployment is valid so missing or incorrect input doesn't stop stop the deployment after a release is
+             * created.
+             *
+             * Note that certain validations still need to be done on the server. Permissions and lifecycle progression
+             * still rely on server side validation.
+             */
+            
+            // Make sure the tags are valid
+            foreach (var tenantTag in TenantTags)
+            {
+                // Verify the format of the tag
+                var parts = tenantTag.Split(Seperator);
+                if (parts.Length != 2 || string.IsNullOrEmpty(parts[0]) || string.IsNullOrEmpty(parts[1]))
+                {
+                    throw new CommandException(
+                        $"Canonical Tag Name expected in the format of `TagSetName{Seperator}TagName`");
+                }
+                // Verify the presence of the tag
+                var tagSets = await Repository.TagSets.FindByName(parts[0]).ConfigureAwait(false);
```

Comment:
This should be done outside the loop otherwise we can be retrieving the same tag more than once

Reply:
And this one @mcasperson ?

Reply:
I've just added a change to cache old queries of the same tagset.